roles/lifecycle_environments/tasks/main.yml

```diff
@@ -0,0 +1,15 @@
+---
+- name: 'Create Lifecycle Environments'
+  theforeman.foreman.lifecycle_environment:
+    username: "{{ username }}"
```

Comment:
It would be good to namespace these like `foreman_username` to avoid conflict with other variables. Also, should default to omit so that env vars could be used.

Reply:
We decided not to prefix the common ones (user, pass, url, org), but omit is a good point!

Reply:
I think that leaves a lot of room for variable conflicts. Ansible has a flat namespace so without a prefix that leaves very generic variable names tied specifically to this role. Here is an example of how I intend to use these roles for reference. https://github.com/redhat-cop/agnosticd/blob/development/ansible/configs/smart-management/sample_vars_patching.yml

Reply:
I filed https://github.com/theforeman/foreman-ansible-modules/issues/1117 since the lack of omit is present across various roles.

Reply:
I am assuming we are talking about the variable `username` versus the parameter `username` here, which I agree with willtome.  If you are already using username (e.g. for Ansible Tower, or another box) there can be massive conflicts.  The extra_vars will always win, so if someone is using Ansible Tower survey tool and happened to choose survey they could mess up this role and not understand why.

Reply:
Opened #1118 to continue this discussion